medusa/indexers/tmdb/tmdb.py

```diff
@@ -603,11 +603,10 @@ def get_id_by_external(self, **kwargs):
                 if result.get('tv_results') and result['tv_results'][0]:
                     # Get the external id's for the passed shows id.
                     externals = self.tmdb.TV(result['tv_results'][0]['id']).external_ids()
-                    externals['tmdb_id'] = result['tv_results'][0]['id']
-
                     externals = {external_id: external_value
                                  for external_id, external_value
                                  in externals.items()
                                  if external_value and external_id in ['tvrage_id', 'imdb_id', 'tvdb_id']}
+                    externals['tmdb_id'] = result['tv_results'][0]['id']
```

Comment:
well, i figure it out later that moving this is not needed because only in tvmaze we check the `mapping` list.  and the externals['tvmaze_id'] need to be done after that list compreh.